scarlet/observation.py

```diff
@@ -0,0 +1,233 @@
+                # Create the matching kernel
+                kernel_fft = observed_fft / target_fft
+                # Take the inverse Fourier transform to normalize the result
+                # Trials without this operation are slow to converge, but in the future
+                # we may be able to come up with a method to normalize in the Fourier Transform
+                # and avoid this step.
+                kernel = np.fft.ifft2(kernel_fft)
+                kernel = np.fft.fftshift(np.real(kernel))
+                kernel /= kernel.sum()
+                # Store the Fourier transform of the matching kernel
+                new_kernel_fft.append(np.fft.fft2(np.fft.ifftshift(kernel)))
+            self.psfs_fft = np.array(new_kernel_fft)
+
+        # 4) divide obs.psf from scene.psf in Fourier space
+
+        # [ 5) compute sparse representation of interpolation * convolution ]
+
+    @property
+    def images(self):
+        return self._images
+
+    @property
+    def weights(self):
+        if self._weights is None:
+            return None
+        elif self._weights == 1:
+            return self._weights
+        return self._weights
+
+    def get_model(self, model):
+        """Resample and convolve a model to the observation frame
+
+        Parameters
+        ----------
+        model: array
+            The model in some other data frame.
+
+        Returns
+        -------
+        model: array
+            The convolved and resampled `model` in the observation frame.
+        """
+        if self.wcs is not None:
+            msg = "get_model is currently only supported when the observation frame matches the scene"
+            raise NotImplementedError(msg)
+
+        def _convolve_band(model, psf_fft):
+            """Convolve the model in a single band
+            """
+            _model = np.pad(model, self.image_padding, 'constant')
+            model_fft = np.fft.fft2(np.fft.ifftshift(_model))
+            convolved_fft = model_fft * psf_fft
+            convolved = np.fft.ifft2(convolved_fft)
+            result = np.fft.fftshift(np.real(convolved))
+            (bottom, top), (left, right) = self.image_padding
+            result = result[bottom:-top, left:-right]
+            return result
+        model = np.array([_convolve_band(model[b], self.psfs_fft[b]) for b in range(self.B)])
+        return model
+
+    def get_loss(self, model):
+        """Calculate the loss function for the model
+
+        Parameters
+        ----------
+        model: array
+            The model in some other data frame.
+
+        Returns
+        -------
+        result: array
+            Scalar tensor with the likelihood of the model
+            given the image data.
+        """
+        if self._psfs is not None:
+            model = self.get_model(model)
+        return np.sum(0.5 * (self._weights * (model - self._images))**2)
+
+    def get_scene(self, scene):
```

Comment:
The utility of this method is unclear to me. I know that we talked about its potential uses e.g. to initialize sources, but our conclusion was that it cannot serve its purpose as one would have to do deconvolutions etc

I suggest to remove

Reply:
removed